### PR TITLE
fix(providers): enhancement of adaptation to breaking change in google maps api.

### PR DESCRIPTION
**Pull request by flyingtof**
After integration team feedback, add backward compatibility to preserve
working habits with google map(can still use a div with a css class as
label definition)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -173,8 +173,7 @@ Here are the common options:
             anchor: new google.maps.Point(0, 32)
         },
         label: function(point) {
-            // a function that return the text value of the label
-            return point.data.location.name;
+            return '<div class="map-custom-label">' + point.id + '</div>';
```
Comment (ThibaudAV) [inline README.md:176]:
```
`<div class="map-custom-label">${point.id}</div>` 
``` 
:p ?

**Review by ThibaudAV — APPROVED** [review]

File: src/utils/dom.js [hunk 2]
```diff
@@ -50,5 +50,22 @@ module.exports = {
...
+
+    extractTextAndCssClasses: function (str) {
+        let div = document.createElement('div');
+        div.innerHTML = str;
+        let textContent = div.textContent || div.innerText || "";
+        let classes = [];
+        if ( isObject(div.firstChild) && isObject(div.firstChild.classList)) {
+            for (let i = 0; i < div.firstChild.classList.length; i++) {
+                classes.push(div.firstChild.classList[i]);
+            }
```
Comment (gaetanmaisse) [inline src/utils/dom.js]: Why not `classes = div.firstChild.classList;` instead of for-loop?

File: README.md (hunk 1 above)
Comment (flyingtof) [reply README.md:176]: This expression was as is in previous versions of README.md. I choosed the 'no surprise' strategy to show that there is no breaking change here. But I totally aggree that a template literal would be better here.

File: src/utils/dom.js (hunk 2 above)
Comment (flyingtof) [reply src/utils/dom.js]: It was a bit ugly but div.firstChild.classList is not an array and thus can not be manipulated in a classical way.
I changed the api to return div.firstChild.classList.value which is a simple string that contain all classes assigned in the html frament

File: src/providers/googleMap/Marker.js [hunk 3]
```diff
@@ -33,31 +34,79 @@ class Marker extends google.maps.Marker {
...
 
     if (true === isDefined(options.icon)) {
-        objectAssign(marker, {
-            icon: options.icon
+        let iconOptions = objectAssign({}, options.icon);
+        objectAssign(markerOptions, {
+            icon: iconOptions
+        });
+    }
+    objectAssign(markerOptions, {labelAnchor: new google.maps.Point(20, 20)});
```
Comment (kavrillon) [inline src/providers/googleMap/Marker.js]: `20`: the anchor depends of the dimensions of the marker, no ?

**Review by kavrillon — APPROVED** [review]

File: src/providers/googleMap/Marker.js (hunk 3 above)
Comment (flyingtof) [reply src/providers/googleMap/Marker.js]: My bad! It is a residual code from tuning phase. I removed this line

**Pull request merged**